tf-modules/cumulus/distribution.tf

```diff
@@ -1,31 +1,33 @@
-module "distribution" {
-  source = "../distribution"
+# Uncomment if using TEA for distribution
```

Comment:
This file and `tf-modules/cumulus/outputs.tf` might actually need to be uncommented as we're still intending to support TEA as the default distribution in `cumulus-template-deploy`. If this module and it's outputs are removed here I think that would break the template. If we uncomment this and leave it in we'd always be deploying the distribution module even if we don't use it though.

Reply:
Ideally I think the `cumulus-template-deploy` would just have this change:

https://github.com/nasa/cumulus-template-deploy/pull/63/files

This would require uncommenting the `distribution` module and the outputs below. We would be deploying both but maybe that's acceptable.

Reply:
So this file is just including the `distribution` in the `cumulus` module. And the `distribution` module is not deploying TEA, it's just attaching some extra functionality to it, namely the S3 credentials endpoint. So whether the code in here is commented/deleted or not, TEA gets deployed from `example/cumulus/thin_egress_app.tf`.

Still, if people are using TEA, we want to have this functionality enabled.

I would probably go with a different approach here of just making the whole module conditional using a `count`. It looks like `distribution/variables.tf` says that `tea_internal_api_endpoint` is a required variable for that module, so you could make this module conditional on that like so

`count: var.tea_internal_api_endpoint !== null ? 1 : 0`

Reply:
Since we need to support both TEA and cumulus_distribution,  is it possible to deploy both distribution endpoints and point distribution only to cumulus_distribution?  This way, if we need to update `s3credentials` functionalities, we can easily test it out for both distribution endpoints.

Reply:
This is helpful, thanks both for chiming in. 

Mark, to your point it totally makes sense that this doesn't deploy TEA. I also have the same pattern of commenting out everything in `example/cumulus/thin_egress_app.tf` that will also need to change either to a conditional or to always be included.

Jenny, if we left this `distribution` module as it was and always deploy it, would we also need to always deploy `example/cumulus/thin_egress_app.tf`? 

In my mind the options are:

1) Always deploy TEA and Cognito by leaving `tf-modules/cumulus/distribution.tf` and `example/cumulus/thin_egress_app.tf` uncommented. The switch between Cognito and TEA would just be handled in cumulus-tf/main.tf.

2) Conditionally switch both `tf-modules/cumulus/distribution.tf` as Mark was suggesting, and `example/cumulus/thin_egress_app.tf` as well.

Reply:
I like the idea '2. Conditionally switch'

Reply:
3. Per Jenny's suggestion: https://github.com/nasa/cumulus/pull/2300#discussion_r662480391 make these `tf-modules/cumulus/distribution.tf.example` and `example/cumulus/thin_egress_app.tf.example` to not always deploy them but leave them available and obvious.

Reply:
We can't do `tf-modules/cumulus/distribution.tf.example` since it's part of the tf-modules, and users won't update it.

Reply:
If we are intending to support both TEA and cumulus distribution in the next release, which is what I gather from other comments in this PR, then we can't comment the contents of this file. 

Because deploying the `distribution` module is necessary for Cumulus to work with TEA. It creates the S3 credentials endpoint on TEA and the TEA bucket map cache file.

Thus, I think the only option is to make this module conditional on whether TEA seems to be deployed, which maybe we can infer from the presence of the `var.tea_internal_api_endpoint` variable

The question about what to do with `example/cumulus/thin_egress_app.tf` and relates more to whether we want to deploy both TEA and cumulus distribution for our example deployments

Reply:
The latest changes to this PR (sorry, it's spread over a few commits) make `distribution` conditional and `thin_egress_app` remains unchanged and is always deployed. The switch between the two distribution APIs is controlled by main.tf.